persistence/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/OffsetStore.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.scaladsl.persistence
+
+import javax.inject.Singleton
+
+import akka.Done
+import com.lightbend.lagom.scaladsl.persistence.Offset
+
+import scala.collection.concurrent
+import scala.concurrent.Future
+import com.lightbend.lagom.scaladsl.persistence.NoOffset
+
+/**
+ * Offset store implementation
+ */
+trait OffsetStore {
+
+  /**
+   * Prepare this offset store to process the given ID and tag.
+   *
+   * @param eventProcessorId The ID of the event processor.
+   * @param tag The tag to prepare for.
+   * @return The DAO, with the loaded offset.
+   */
+  def prepare(eventProcessorId: String, tag: String): Future[OffsetDao]
+}
+
+/**
+ * A prepared DAO for storing offsets.
+ */
+trait OffsetDao {
+
+  /**
+   * The last offset processed.
+   */
+  val loadedOffset: Offset
+
+  /**
+   * Save the given offset.
+   *
+   * @param offset The offset to save.
+   * @return A future that is redeemed when the offset has been saved.
+   */
+  def saveOffset(offset: Offset): Future[Done]
+}
```

Comment:
I think it's just the Offset type that's specific to the java/scala DSL public APIs, this API could be changed to use it's own type to represent an offset, I think the value of the code that can be shared because of that will be worth it.